src/yachtzee/index.ts

```diff
@@ -1,163 +1,116 @@
-  if (haveEffect($effect`Beaten Up`)) {
-    uneffect($effect`Beaten Up`);
-  }
-  meatMood(false, 2000).execute(Math.min(jellyTurns, fishyTurns));
-  safeRestore();
+  return [
+    {
+      name: "Yachtzee",
+      ready: () => get("noncombatForcerActive") && have($effect`Fishy`),
+      completed: () => false,
+      do: $location`The Sunken Party Yacht`,
+      outfit: () => {
+        return {
+          familiar,
 
-  propertyManager.setChoice(918, 2);
-  let plantCrookweed = true;
-  while (Math.min(jellyTurns, fishyTurns) > 0) {
-    executeNextDietStep();
-    if (!get("noncombatForcerActive")) throw new Error("We did not use stench jellies");
-    // Switch familiars in case changes in fam weight from buffs means our current familiar is no longer optimal
-    prepareOutfitAndFamiliar();
-    if (!have($effect`Really Deep Breath`)) {
-      const bestWaterBreathingEquipment = getBestWaterBreathingEquipment(
-        Math.min(jellyTurns, fishyTurns),
-      );
-      if (bestWaterBreathingEquipment.item !== $item.none) {
-        equip(bestWaterBreathingEquipment.item);
-      }
-      if (
-        haveEquipped($item`The Crown of Ed the Undying`) &&
-        !booleanModifier("Adventure Underwater")
-      ) {
-        cliExecute("edpiece fish");
-      }
-    }
-    if (!have($effect`Polka of Plenty`)) {
-      if (have($effect`Ode to Booze`)) cliExecute(`shrug ${$effect`Ode to Booze`}`);
-      if (
-        getActiveSongs().length < (have($skill`Mariachi Memory`) ? 4 : 3) &&
-        have($skill`The Polka of Plenty`)
-      ) {
-        useSkill($skill`The Polka of Plenty`);
-      }
-    }
-    garboAdventure($location`The Sunken Party Yacht`, Macro.abort());
-    if (get("lastEncounter") === "Yachtzee!") eventLog.yachtzees += 1;
-    if (myTurncount() > turncount || haveEffect($effect`Fishy`) < fishyTurns) {
-      fishyTurns -= 1;
-      jellyTurns -= 1;
-      turncount = myTurncount();
-      set("_stenchJellyChargeTarget", get("_stenchJellyChargeTarget", 0) - 1);
-    }
-    if (plantCrookweed && FloristFriar.have() && FloristFriar.Crookweed.available()) {
-      FloristFriar.Crookweed.plant();
-    }
-    plantCrookweed = false;
-    postCombatActions();
+          ...meatOutfit,
+        };
+      },
+      post: () => {
+        if (
+          canAdventure($location`The Spooky Forest`) &&
+          FloristFriar.have() &&
+          FloristFriar.Crookweed.available()
+        ) {
+          FloristFriar.Crookweed.plant();
+        }
+      },
+    },
+    {
+      name: "Fishy Pipe",
+      ready: () => have($item`fishy pipe`) && !have($effect`Fishy`),
+      completed: () => get("_fishyPipeUsed"),
+      do: () => use($item`fishy pipe`),
+    },
+    {
+      name: "Spikolodon Spikes",
+      ready: () => have($item`Jurassic Parka`) && get("_spikolodonSpikeUses") < 5,
+      completed: () => get("noncombatForcerActive"),
+      do: () => {
+        const run = tryFindFreeRun(freeRunConstraints(false)) ?? ltbRun();
```

Comment:
imo we should make this a `let` variable outside, initialize it as part of the start of the task and dump it at the end. that way we can go through traditional outfit and combat code for grimoire. freecandy does this [here](https://github.com/loathers/freecandydotexe/blob/ef62f543e34330a9467ce4d9a614f0db85def5ee/src/regularTasks.ts#L74-L89), and it gets used [here](https://github.com/loathers/freecandydotexe/blob/ef62f543e34330a9467ce4d9a614f0db85def5ee/src/regularTasks.ts#L316-L343)